code/renderers/react/src/entry-preview-rsc.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import semver from 'semver';
+import type { Addon_DecoratorFunction } from '@storybook/types';
+import type { StoryContext } from './types';
+
+export const ServerComponentDecorator = (
+  Story: React.FC,
+  { parameters }: StoryContext
+): React.ReactNode => {
+  if (!parameters?.react?.rsc) return <Story />;
+
+  if (semver.major(React.version) < 18 || semver.minor(React.version) < 3) {
```

Comment:
This condition returns true for a hypothetical upcoming 19.0 version and will throw an error.